File: src/commonMain/kotlin/me/kolterdyx/compiler/statement/ExpressionStatement.kt
package me.kolterdyx.compiler.statement

import me.kolterdyx.compiler.ast.StatementVisitor
import me.kolterdyx.compiler.expression.Expression

class ExpressionStatement(
    val expression: Expression
) : Statement() {
    override fun <R> accept(visitor: StatementVisitor<R>): R {
        return visitor.visitExpression(this)
    }
}

File: internal/ast/expressionStatement.go
package ast

import "github.com/Kolterdyx/mcbasic/internal/interfaces"

type ExpressionStmt struct {
	Expression Expr
}

func (e ExpressionStmt) Accept(visitor StatementVisitor) any {
	return visitor.VisitExpression(e)
}

func (e ExpressionStmt) Type() NodeType {
	return ExpressionStatement
}

func (e ExpressionStmt) ToString() string {
	return e.Expression.ToString() + ";"
}

func (e ExpressionStmt) GetSourceLocation() interfaces.SourceLocation {
	return e.Expression.GetSourceLocation()
}
